### PR TITLE
Feature/path targeting

### PR DESCRIPTION
**Pull request by nvh**

### CODE REVIEW

File: framer/Animation.coffee [hunk 1]
```diff
@@ -417,7 +403,7 @@ class exports.Animation extends BaseClass
 		return _.pick(@layer, _.keys(@properties))
 
 	@isAnimatable = (v) ->
-		_.isNumber(v) or _.isFunction(v) or isRelativeProperty(v) or Color.isColorObject(v) or Gradient.isGradientObject(v) or SVGPath.isPath(v)
+		_.isNumber(v) or _.isFunction(v) or isRelativeProperty(v) or Color.isColorObject(v) or Gradient.isGradientObject(v) #or SVGPath.isPath(v)
```
Comment (eelco) [inline framer/Animation.coffee]: Remove commented out code.

File: framer/SVGPath.coffee [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		path._properties.strokeFraction = value / path.length
+		if _.isEmpty path.strokeDasharray
+			path.strokeDasharray = [path.length]
+		path.strokeDashoffset = path.length - value
+	@define "strokeFraction", layerProperty @, "strokeFraction", null, null, _.isNumber, null, {}, (path, value) ->
+		path.strokeLength = path.length * value
+
+	@define "length",
+		get: ->
+			@_length
```
Comment (eelco) [inline framer/SVGPath.coffee]: I think this is more readable as a one-liner

File: framer/Animation.coffee [hunk 3]
```diff
@@ -280,21 +279,7 @@ class exports.Animation extends BaseClass
 	_prepareUpdateValues: =>
 		@_valueUpdaters = {}
 		for k, v of @_stateB
-			if SVGPath.isPath(v)
```
Comment (eelco) [inline framer/Animation.coffee]: What does removing this break?

Comment (nvh) [reply framer/Animation.coffee]: Ah yes, that is part that is not done yet. It breaks animation along a path

File: framer/Animation.coffee (hunk 1 above)
Comment (nvh) [reply framer/Animation.coffee]: Yeah it should be enabled again, but I had some file-inheritance/require issues that I solved by disabling this alltogether

File: framer/SVGGroup.coffee [hunk 4]
```diff
@@ -1,59 +1,59 @@
 {Layer, layerProperty} = require "./Layer"
 {Color} = require "./Color"
 {SVG} = require "./SVG"
+
+{SVGBaseLayer} = require "./SVGBaseLayer"
 {SVGPath} = require "./SVGPath"
 
-class SVGGroup extends Layer
+areEqual = (a, b) ->
```
Comment (eelco) [inline framer/SVGGroup.coffee]: This is an oddly specific equality check. What’s not in lodash / already in Framer that requires this?

File: framer/SVGGroup.coffee [hunk 5]
```diff
@@ -66,23 +66,8 @@ class SVGGroup extends Layer
 
 	@defineGroupProxyProp "fill"
 	@defineGroupProxyProp "stroke"
-	@defineGroupProxyProp "strokeWidth", _.isNumber, parseInt, Object.prototype.toString
-	@defineGroupProxyProp "strokeWidthMultiplier", _.isNumber, parseInt, Object.prototype.toString
+	@defineGroupProxyProp "strokeWidth", _.isNumber, parseInt
+	@defineGroupProxyProp "strokeWidthMultiplier", _.isNumber, parseInt
 	@defineGroupProxyProp "color", Color.validColorValue, Color.toColor
 
-	@define "gradient",
```
Comment (eelco) [inline framer/SVGGroup.coffee]: No gradient on groups? (Is it part of the fill/stroke?)

File: framer/Layer.coffee [hunk 6]
```diff
@@ -178,6 +178,7 @@ class exports.Layer extends BaseClass
 
 		# Sanitize calculated property setters so direct properties always win
 		layerPropertyIgnore(options, "point", ["x", "y"])
+		layerPropertyIgnore(options, "midPoint", ["midX", "midY"])
```
Comment (eelco) [inline framer/Layer.coffee]: Where is this used?

Comment (nvh) [reply framer/Layer.coffee]: Nowhere, it's public api to be able to do
`layer.midPoint = path.start`

File: framer/SVGGroup.coffee (hunk 4 above)
Comment (nvh) [reply framer/SVGGroup.coffee]: Fixed by creating `Utils.equal`

File: framer/SVGGroup.coffee (hunk 5 above)
Comment (nvh) [reply framer/SVGGroup.coffee]: Currently we don't support gradients on shapes at all, we will implement this in the future.

**Pull request merged**